File: kora-app-annotation-processor/src/test/java/io/koraframework/kora/app/annotation/processor/app/AppWithModuleOf.java
package io.koraframework.kora.app.annotation.processor.app;

import io.koraframework.common.KoraApp;
import io.koraframework.common.annotation.Root;

@KoraApp
public interface AppWithModuleOf {
    @Root
    default Class1 class1(Class2 class2) {
        return new Class1(class2);
    }

    record Class1(Class2 class2) {}

    record Class2() {}

    record Class3() {}

    default Class3 class2() {
        return new Class3();
    }

    @io.koraframework.common.Module
    interface Module {
        default Class2 class2() {
            return new Class2();
        }
    }

}

File: kora-app-symbol-processor/src/test/kotlin/io/koraframework/kora/app/ksp/app/AppWithModuleOf.kt
package io.koraframework.kora.app.ksp.app

import io.koraframework.common.KoraApp
import io.koraframework.common.annotation.Root

@KoraApp
interface AppWithModuleOf {
    @Root
    fun class1(class2: Class2): Class1 {
        return Class1(class2)
    }

    fun class2(): Class3 {
        return Class3()
    }

    data class Class1(val class2: Class2)
    open class Class2
    class Class3

    @io.koraframework.common.Module
    interface Module {
        fun class2(): Class2 {
            return Class2()
        }
    }
}
